### DISCUSSION TITLE
Welcome to Super-Delaunay Discussions!

### DISCUSSION THREAD

**Discussion by DanielAbalde** [DanielAbalde/Super-Delaunay#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by LaVieon** [comment]
Very Fantastic Plugins！Likes it a Lot！

**Comment by hisaid** [comment]
Hi Daniel. I am using Super Delaunay for floorplan generation optimization (using Octopus). The decision variables are the room locations and the room weights (sizes). The Voronoi weights cause a lot of instability in the model if left to be generatively configured, the Voronoi mesh is either not generated or is created with cells less than the number of rooms (points). Is there a way to control a the weight values impact on the created Voronoi?

**Reply by DanielAbalde** [reply]
You need to balance the weights around neighbours to avoid cell colapse, but there is no easy way to do it. You have to penalise this undesired behaviour in your fitness function.

**Reply by hisaid** [reply]
Thanks Daniel for the reply. To implement the balancing script, can you share more info on the weighted Voronoi algorithm? The[ link on the Super Delaunay page](https://en.wikipedia.org/wiki/Weighted_Voronoi_diagram) listed different algorithm types.

**Reply by DanielAbalde** [reply]
Sorry but you will not be able without making your own implementation (and even so, it will be extremely difficult).

Here you have the different distance metrics: https://github.com/DanielAbalde/PowerDiagrams/blob/3217ca45f01052111113ed9b37f0bb9f04f213d8/Delaunay2d.cs#L2415 
Here you have the code of the algorithm plus extra info:
https://github.com/DanielAbalde/PowerDiagrams/blob/3217ca45f01052111113ed9b37f0bb9f04f213d8/Delaunay2d.cs#L315

The result of each cell depends on the relation between this custom distances of all their neighbours, the algorithm and the order of the points being added.

Again, I recommend you to penalise the bad behaviour (like collapsing cells or too small or too big) by giving a very high (or low depending on whether you maximise or minimise) result to the fitness function when this happen.

**Reply by hisaid** [reply]
Thanks Daniel, that was super helpful
